contributor_docs/fes_reference_dev_notes.md

```diff
@@ -0,0 +1,456 @@
+````
+Example of an Error in User's Sketch (Scope)
+````JavaScript
+function setup() {
+  let b = 1;
+}
+function draw() {
+  b += 1;
+}
+// FES will show:
+// > 🌸 p5.js says: There's an error due to "b" not being defined in the current scope (on line 5 in sketch.js [http://localhost:8000/lib/empty-example/sketch.js:5:3]). If you have defined it in your code, you should check its scope, spelling, and letter-casing (JavaScript is case-sensitive). For more: https://p5js.org/examples/data-variable-scope.html https://developer.mozilla.org/docs/Web/JavaScript/Reference/Errors/Not_Defined#What_went_wrong
+````
+Example of an Error in User's Sketch Example (Spelling)
+````JavaScript
+function setup() {
+  colour(1, 2, 3);
+}
+// FES will show:
+// > 🌸 p5.js says: It seems that you may have accidentally written "colour" instead of "color" (on line 2 in sketch.js [http://localhost:8000/lib/empty-example/sketch.js:2:3]). Please correct it to color if you wish to use the function from p5.js (http://p5js.org/reference/#/p5/color)
+````
+##### Description
+`fesErrorMonitor()` handles various errors that the browser shows. The function generates global error messages.
+
+Generates and prints...
+* ... a friendly error message using key: `fes.globalErrors.syntax.*`, `fes.globalErrors.reference.*`, `fes.globalErrors.type.*`.
+* ... an "internal library" error message via `processStack()` using key: `fes.wrongPreload`, `fes.libraryError`.
+* ... a stacktrace message via `printFriendlyStack()` using key: `fes.globalErrors.stackTop`,`fes.globalErrors.stackSubseq`.
+* ... a spell-check message (from a reference error) via `handleMisspelling()` using key: `fes.misspelling`.
+
+`_fesErrorMonitor()` can be called either by an error event, an unhandled rejection event, or it can be manually called in a catch block as follows:
+```
+try { someCode(); } catch(err) { p5._fesErrorMonitor(err); }
+```
+
+The function currently works with some kinds of ReferenceError, SyntaxError, and TypeError. You can find the complete list of supported errors in `browser_errors.js`.
+
+The call sequence for `_fesErrorMonitor` roughly looks something like this:
+```
+ _fesErrorMonitor
+     processStack
+       printFriendlyError
+     (if type of error is ReferenceError)
+       _handleMisspelling
+         computeEditDistance
+         _report
+       _report
+       printFriendlyStack
+     (if type of error is SyntaxError, TypeError, etc)
+       _report
+       printFriendlyStack
+```
+##### Syntax
+````JavaScript
+fesErrorMonitor(event)
+````
+##### Parameters
+```
+@param {*}  e     Error event
+```
+##### Location
+core/friendly_errors/fes_core.js
+
+### `fesCodeReader()`
+##### Examples
+Example of Redefining p5.js Reserved Constant
+````JavaScript
+function setup() {
+  //PI is a p5.js reserved constant
+  let PI = 100;
+}
+// FES will show:
```

Comment:
These are great examples. I've noticed here & elsewhere that this messages are long enough to cause a horizontal scroll when displayed in GitHub in the README. I don't know if there's anyway we can wrap the text instead, but it would be nice if it did. (Feel free to ignore this comment if it adds too much work)

Reply:
This is a good suggestion. I will pull out the FES output preview messages from the code blocks and put them into quotes instead. Quote should support word-wrap by default.